src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int INITIAL_VALUE = 0;
+    private static final int OPERATION_PART = 0;
+    private static final int TRANSACTION_AMOUNT_PART = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supply = INITIAL_VALUE;
+        int buy = INITIAL_VALUE;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String csvData;
```

Parsing code uses `csvData.split(",")` and immediately indexes into `dataParts` and calls `Integer.parseInt`. The checklist requires predictable handling of malformed input and proper exception flow: "Don't ignore exceptions". Validate that `dataParts.length == 2` before indexing and catch `NumberFormatException` to rethrow a `RuntimeException` with a clear message and the original cause. Do not allow ArrayIndexOutOfBoundsException or NumberFormatException to propagate without context.